app/renderer/components/Session/AttachToSession.js

```diff
@@ -25,26 +25,33 @@ function formatCapsBrowserstack (caps) {
 }
 
 function formatCapsLambdaTest (session) {
-  let caps;
-  if (session.capabilities) {
+  let caps = session, deviceName = session.deviceName;
+  if (Object.keys(session.capabilities).length !== 0) {
     caps = session.capabilities;
+    deviceName = session.capabilities.desired ? session.capabilities.desired.deviceName : session.capabilities.deviceName;
   } else if (session.desired) {
     caps = session.desired;
-  } else {
-    caps = session;
+    deviceName = session.desired.deviceName;
   }
-  let importantCaps = [caps.deviceName, caps.platformName, caps.platformVersion];
+  let importantCaps = [deviceName, caps.platformName, caps.platformVersion];
   return importantCaps.join(', ').trim();
 }
 
 export default class AttachToSession extends Component {
 
+  getSessionId (session) {
+    return session.sessionId === undefined || session.sessionId === 'undefined' ? session.id : session.sessionId;
```

Comment:
What situations an be `'undefined'` string? So far, Appium's `/sessions` endpoint returns the list of session id (as `id`) and `capabilities`. So we need to understand this `sessionId`.

Reply:
Fixed this